assets/js/controllers/sfox/sfoxLink.controller.js

```diff
@@ -72,5 +74,56 @@ function SfoxLinkController ($scope, $q, sfox, modals) {
       .finally($scope.free);
   };
 
+  $scope.getBankAccounts = (token) => {
+    $scope.token = token;
+    $q.resolve(exchange.bankLink.getAccounts($scope.token))
+      .then((bankAccounts) => $scope.state.bankAccounts = bankAccounts)
+      .then(() => $scope.fields.bankAccount = $scope.state.bankAccounts[0])
+      .catch(sfox.displayError);
+  };
+
+  $scope.setBankAccount = () => {
+    let account;
+
+    let addAccount = (methods) => methods.ach.addAccount(
+      account.routing_number,
+      account.account_number,
+      account.bank_name,
+      $scope.fields.bankAccount.subtype
+    );
+
+    let obj = {
+      token: $scope.token,
+      id: $scope.fields.bankAccount._id,
+      lastName: exchange.profile.lastName || null,
+      firstName: exchange.profile.firstName || null
+    };
+
+    $q.resolve(exchange.bankLink.setAccount(obj))
+      .then((res) => account = res.bankAccount)
+      .then(exchange.getBuyMethods()
+      .then((methods) => addAccount))
+      .then(() => $scope.vm.goTo('buy'))
+      .catch(sfox.displayError)
+      .finally($scope.free);
+  };
```

Comment:
There's some weird nesting of promises here, I think it would be better to keep them in a single chain. Also, doesn't look like `addAccount` is ever called.